src/tail/mod.rs

```diff
@@ -0,0 +1,56 @@
+mod host;
+mod session;
+mod tunnel;
+
+use host::Host;
+use session::Session;
+use tunnel::Tunnel;
+
+use tokio;
+use tokio::runtime::Runtime as TokioRuntime;
+use tokio::sync::oneshot;
+
+use crate::settings::global_user::GlobalUser;
+use crate::settings::toml::Target;
+
+pub struct Tail;
+
+impl Tail {
+    pub fn run(target: Target, user: GlobalUser) -> Result<(), failure::Error> {
+        let mut runtime = TokioRuntime::new()?;
+
+        runtime.block_on(async {
+            let (log_tx, log_rx) = oneshot::channel();
+            let log_host = Host::new()?;
+            let host_handle = tokio::spawn(log_host.run(log_rx));
+
+            let tunnel_process = Tunnel::new()?;
+            let (tunnel_tx, tunnel_rx) = tokio::sync::oneshot::channel();
+            let tunnel_handle = tokio::spawn(tunnel_process.run(tunnel_rx));
+
+            let (session_tx, session_rx) = tokio::sync::oneshot::channel();
+            let session_handle = tokio::spawn(Session::run(target, user, session_rx));
+
+            let txs = vec![log_tx, tunnel_tx, session_tx];
+            let sigint_handle = tokio::spawn(handle_sigint(txs));
+
```

Comment:
TODO: receiving a ctrl-c before the message "Ready to stream logs" hangs. need to handle that problem.